### PR TITLE
ZEPPELIN-2079. Upgrade livy to 0.3 in livy interpreter

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
Upgrade livy to 0.3.

* Add new tests for livy 0.3
* Add 2 livy build in travis. (livy 0.2 + spark 1.6.3 and livy 0.3 + spark 2.1.0, unfortunately livy 0.3 has some packaging issue which cause some issues for integration test for livy 0.3 + spark 1.6.3). I also merge the livy build into spark build in travis but has to set `sudo` as `required` for more memory. 

### What type of PR is it?
[Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-2079

### How should this be tested?
Tested is added

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 language: java
 
-sudo: false
+sudo: required
```
Comment (felixcheung) [inline .travis.yml]: why do we need this change (it's a big change)

**Comment by felixcheung** [comment]
@zjffdu could you please clean up the templated text from the PR description box?

**Comment by zjffdu** [comment]
Sorry, forget to add WIP.

**Comment by zjffdu** [comment]
@felixcheung @prabhjyotsingh @del680202 Please help review.

File: .travis.yml [hunk 2]
```diff
@@ -41,16 +41,18 @@ matrix:
       env: SCALA_VER="2.11" SPARK_VER="2.0.2" HADOOP_VER="2.6" PROFILE="-Prat" BUILD_FLAG="clean" TEST_FLAG="org.apache.rat:apache-rat-plugin:check" TEST_PROJECTS=""
 
     # Test all modules with spark 2.1.0 and scala 2.11
-    - jdk: "oraclejdk7"
-      env: SCALA_VER="2.11" SPARK_VER="2.1.0" HADOOP_VER="2.6" PROFILE="-Pspark-2.1 -Phadoop-2.6 -Ppyspark -Psparkr -Pscalding -Phelium-dev -Pexamples -Pscala-2.11" BUILD_FLAG="package -Pbuild-distr -DskipRat" TEST_FLAG="verify -Pusing-packaged-distr -DskipRat" TEST_PROJECTS=""
+    - sudo: required
```
Comment (felixcheung) [inline .travis.yml]: I guess we still have `sudo: required` here - could you elaborate why this is needed?

Comment (zjffdu) [inline .travis.yml]: This is for Livy integration test which require more memory. See https://docs.travis-ci.com/user/ci-environment/

**Comment by zjffdu** [comment]
ping @felixcheung @prabhjyotsingh @del680202

**Comment by prabhjyotsingh** [comment]
@zjffdu  I'll try to run and review this today.

File: .travis.yml (hunk 2 above)
Comment (felixcheung) [inline .travis.yml]: I guess it's be great to get more eyes on the `sudo: required` change. @Leemoonsoo

**Comment by prabhjyotsingh** [comment]
Tried to run these test on Mac 10.12.4  and Cento 6.4 couple of times, and test passed. So, I'll say LGTM!

File: .travis.yml (hunk 2 above)
Comment (zjffdu) [inline .travis.yml]: @Leemoonsoo  Do you have bandwidth to review the travis change ? Thanks

Comment (Leemoonsoo) [inline .travis.yml]: Adding `sudo: required` looks fine.

**Comment by felixcheung** [comment]
ok then. 
merging if no more comment

**Comment by felixcheung** [comment]
merged to master.

**Pull request closed without merging**